src/Compilers/Core/Portable/Serialization/ObjectReader.cs

```diff
@@ -100,10 +101,24 @@ internal sealed partial class ObjectReader : IDisposable
                 return null;
             }
 
-            if (stream.ReadByte() != VersionByte1 ||
-                stream.ReadByte() != VersionByte2)
+            try
             {
-                return null;
+                if (stream.ReadByte() != VersionByte1 ||
+                    stream.ReadByte() != VersionByte2)
+                {
+                    return null;
+                }
+            }
+            catch (AggregateException ex) when (ex.InnerException is not null)
+            {
+                // PipeReaderStream wraps any exception it throws in an AggregateException, which is not expected by
+                // callers treating it as a normal stream. Unwrap and rethrow the inner exception for clarity.
+                // https://github.com/dotnet/runtime/issues/70206
```

Comment:
>https://github.com/dotnet/runtime/issues/70206 [](http://example.com/codeflow?start=19&length=46)

Do we still need to catch `AggregateException` since the underlying issue is being addressed?

Reply:
➡️ I believe we should fix it unless we can guarantee the runtime environment will adopt this fix in the same time frame. The corresponding feedback ticket has more than 10 duplicates and is marked as a P0 issue.